app/assets/javascripts/quby/answers/validation.js

```diff
@@ -189,6 +208,59 @@
     return !failed;
   };
 
+  function parsePartialDate(inputs) {
+    var values = partialDateValues(inputs)
+    if(!dateValuesValid(values)) {
+      throw "invalidDate";
+    }
+    else {
+      return new Date(values.year, values.month, values.day, values.hour, values.minute);
+    }
+  }
+
+  function dateValuesValid(values) {
+    return values['year']   >= 1900 && values['year']   <= 2100 &&
+           values['month']  >= 0    && values['month']  <= 11   &&
+           values['day']    >= 1    && values['day']    <= 31   &&
+           values['hour']   >= 0    && values['hour']   <= 23   &&
+           values['minute'] >= 0    && values['minute'] <= 59;
+  }
+
+  function partialDateValues(inputs) {
```

Comment:
Het is inderdaad wat duidelijker om te benadrukken dat er defaults worden gebruikt om een "partial date" mogelijk te maken. Ik maak er `dateValuesWithDefaults` van om consistent te zijn met de rest van de functies.